eZ/Publish/Core/Repository/TrashService.php

```diff
@@ -130,9 +130,13 @@ public function trash(Location $location)
             throw $e;
         }
 
-        return isset($spiTrashItem)
-            ? $this->buildDomainTrashItemObject($spiTrashItem)
-            : null;
+        try {
+            return isset($spiTrashItem)
+                ? $this->buildDomainTrashItemObject($spiTrashItem)
+                : null;
+        } catch (UnauthorizedException $e) {
+            return true;
+        }
```

Comment:
If you're gonna like my suggestion, then it should be handled & set inside `buildDomainTrashItemObject` due to responsibility.

Reply:
No longer relevant, using sudo instead.